internal/auth/store/session/store_memory.go

```diff
@@ -139,82 +137,81 @@ func (s *InMemorySessionStore) ListAll(_ context.Context) (map[id.SessionID]*mod
-func (s *InMemorySessionStore) AdvanceLastSeen(_ context.Context, sessionID id.SessionID, clientID string, at time.Time, accessTokenJTI string, activate bool, deviceID string, deviceFingerprintHash string) (*models.Session, error) {
+// It validates the session using domain logic, then applies updates via domain methods.
+// Errors are returned as sentinel errors per store boundary contract.
+func (s *InMemorySessionStore) AdvanceLastSeen(_ context.Context, sessionID id.SessionID, clientID id.ClientID, at time.Time, accessTokenJTI string, activate bool, deviceID string, deviceFingerprintHash string) (*models.Session, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
 	session, ok := s.sessions[sessionID]
 	if !ok {
 		return nil, fmt.Errorf("session not found: %w", sentinel.ErrNotFound)
 	}
-	if err := s.validateForAdvance(session, clientID, at, true); err != nil {
+
+	// Validate using domain checks, return sentinel errors per store contract
+	if err := validateForAdvance(session, clientID, at, true); err != nil {
 		return nil, err
 	}
 
-	if at.After(session.LastSeenAt) {
-		session.LastSeenAt = at
+	// Apply updates using domain methods
+	session.RecordActivity(at)
+	if activate {
+		session.Activate()
 	}
-	if activate && session.Status == models.SessionStatusPendingConsent {
-		session.Status = models.SessionStatusActive
-	}
-	applyDeviceFields(session, accessTokenJTI, deviceID, deviceFingerprintHash)
+	session.ApplyTokenJTI(accessTokenJTI)
+	session.ApplyDeviceInfo(deviceID, deviceFingerprintHash)
 
 	s.sessions[sessionID] = session
 	return session, nil
 }
 
-func (s *InMemorySessionStore) AdvanceLastRefreshed(_ context.Context, sessionID id.SessionID, clientID string, at time.Time, accessTokenJTI string, deviceID string, deviceFingerprintHash string) (*models.Session, error) {
+// AdvanceLastRefreshed updates the session's refresh timestamp and activity time.
+// Errors are returned as sentinel errors per store boundary contract.
+func (s *InMemorySessionStore) AdvanceLastRefreshed(_ context.Context, sessionID id.SessionID, clientID id.ClientID, at time.Time, accessTokenJTI string, deviceID string, deviceFingerprintHash string) (*models.Session, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
 	session, ok := s.sessions[sessionID]
 	if !ok {
 		return nil, fmt.Errorf("session not found: %w", sentinel.ErrNotFound)
 	}
-	if err := s.validateForAdvance(session, clientID, at, false); err != nil {
+
+	// Validate using domain checks, return sentinel errors per store contract
+	if err := validateForAdvance(session, clientID, at, false); err != nil {
 		return nil, err
 	}
 
-	if session.LastRefreshedAt == nil || at.After(*session.LastRefreshedAt) {
-		session.LastRefreshedAt = &at
-	}
-	if at.After(session.LastSeenAt) {
-		session.LastSeenAt = at
-	}
-	applyDeviceFields(session, accessTokenJTI, deviceID, deviceFingerprintHash)
+	// Apply updates using domain methods
+	session.RecordRefresh(at)
```

Comment:
General comment for stores - should they be changing entity state, or should this happen in service first then store remains pure I/o